File: src/main/java/com/ternbusty/takoyaki/command/StateCommand.java
package com.ternbusty.takoyaki.command;

import com.ternbusty.takoyaki.state.State;
import com.ternbusty.takoyaki.util.Json;

public final class StateCommand {
    private StateCommand() {}

    public static int run(String rootPath, String containerId) {
        try {
            State s = State.load(rootPath, containerId).refreshStatus();
            System.out.println(Json.encode(s.toJson()));
            return 0;
        } catch (Exception e) {
            System.err.println("container " + containerId + " does not exist");
            return 1;
        }
    }
}

File: src/main/java/com/ternbusty/takoyaki/command/StateCommand.kt
package com.ternbusty.takoyaki.command

import com.ternbusty.takoyaki.state.State
import com.ternbusty.takoyaki.util.JsonCodec
object StateCommand {

    fun run(rootPath: String, containerId: String): Int {
        return try {
            val s = State.load(rootPath, containerId).refreshStatus()
            println(JsonCodec.encode(s))
            0
        } catch (e: Exception) {
            System.err.println("container $containerId does not exist")
            1
        }
    }
}
